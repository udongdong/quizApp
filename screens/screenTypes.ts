import {NavigatorScreenParams} from '@react-navigation/native';

export type MainStackParamList = {
  quizScreen: undefined;
  mainTab: NavigatorScreenParams<MainTabParamList>;
};

export type MainTabParamList = {
  quiz: undefined;
  history: undefined;
  review: undefined;
};
