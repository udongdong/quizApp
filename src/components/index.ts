export * from './Button';
export * from './ProgressBar';
export * from './RadioButton';
export * from './Space';
export * from './quiz';
