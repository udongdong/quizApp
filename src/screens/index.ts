export * from './MainScreen';
export * from './MainTab';
export * from './quiz';
export * from './review';
export * from './screenTypes';
