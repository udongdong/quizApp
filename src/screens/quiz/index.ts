export * from './QuizMain';
export * from './QuizResultScreen';
export * from './quizScreen';
