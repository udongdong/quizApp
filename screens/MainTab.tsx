import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import React from 'react';
import {MainTabParamList} from './screenTypes';
import QuizMain from './quiz/QuizMain';

export default function MainTab(): React.JSX.Element {
  const Tab = createBottomTabNavigator<MainTabParamList>();

  return (
    <Tab.Navigator>
      <Tab.Screen name="quiz" component={QuizMain} />
    </Tab.Navigator>
  );
}
