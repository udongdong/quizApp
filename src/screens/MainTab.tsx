import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import React from 'react';
import {MainTabParamList} from './screenTypes';
import QuizMain from './quiz/QuizMain';
import ReviewMain from './review/ReviewMain';
import {color} from '../styles';

export default function MainTab(): React.JSX.Element {
  const Tab = createBottomTabNavigator<MainTabParamList>();

  return (
    <Tab.Navigator
      screenOptions={{
        tabBarLabelPosition: 'beside-icon',
        tabBarLabelStyle: {
          fontWeight: '700',
          fontSize: 15,
        },
        tabBarIconStyle: {display: 'none'},
        tabBarStyle: {height: 60},
        tabBarActiveTintColor: color.primary,
        headerTitleStyle: {
          color: color.primary,
        },
      }}>
      <Tab.Screen
        name="quiz"
        component={QuizMain}
        options={{tabBarLabel: '퀴즈', title: '퀴즈 풀기'}}
      />
      <Tab.Screen
        name="review"
        component={ReviewMain}
        options={{tabBarLabel: '리뷰', title: '퀴즈 리뷰'}}
      />
    </Tab.Navigator>
  );
}
