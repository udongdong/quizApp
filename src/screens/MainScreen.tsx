import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {MainStackParamList} from './screenTypes';
import {MainTab, ReviewDetail, QuizScreen} from './';
import {QuizResultScreen} from './quiz';

export function MainScreen(): React.JSX.Element {
  const MainStack = createNativeStackNavigator<MainStackParamList>();

  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="mainTab"
        screenOptions={{headerShown: false}}>
        <MainStack.Screen name="mainTab" component={MainTab} />
        <MainStack.Screen name="quizScreen" component={QuizScreen} />
        <MainStack.Screen
          name="quizResultScreen"
          component={QuizResultScreen}
        />
        <MainStack.Screen name="reviewDetail" component={ReviewDetail} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
}
