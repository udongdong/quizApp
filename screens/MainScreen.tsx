import {createNativeStackNavigator} from '@react-navigation/native-stack';
import React from 'react';
import {MainStackParamList} from './screenTypes';
import {NavigationContainer} from '@react-navigation/native';
import MainTab from './MainTab';
import QuizScreen from './quiz/QuizScreen';
import {QuizResultScreen} from './quiz/QuizResultScreen';
import ReviewDetail from './review/ReviewDetail';

export default function MainScreen(): React.JSX.Element {
  const MainStack = createNativeStackNavigator<MainStackParamList>();

  return (
    <NavigationContainer>
      <MainStack.Navigator
        initialRouteName="mainTab"
        screenOptions={{headerShown: false}}>
        <MainStack.Screen name="mainTab" component={MainTab} />
        <MainStack.Screen name="quizScreen" component={QuizScreen} />
        <MainStack.Screen
          name="quizResultScreen"
          component={QuizResultScreen}
        />
        <MainStack.Screen name="reviewDetail" component={ReviewDetail} />
      </MainStack.Navigator>
    </NavigationContainer>
  );
}
