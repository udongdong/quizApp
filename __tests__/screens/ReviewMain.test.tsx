import 'react-native';
import React from 'react';
import {ReviewMain} from '../../src/screens';

import {render, screen} from '@testing-library/react-native';
import {MockTabNavigator} from './MockNavigator';
import {MainTabParamList} from '../../src/screens/screenTypes';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

function getComponent() {
  const Tab = createBottomTabNavigator<MainTabParamList>();

  return (
    <MockTabNavigator>
      <Tab.Screen name="review" component={ReviewMain} />
    </MockTabNavigator>
  );
}

describe('ReviewMain test', () => {
  const component = getComponent();

  beforeEach(async () => {
    jest.useFakeTimers();
  });

  test('render ReviewMain', async () => {
    render(component);

    expect(screen).toMatchSnapshot();
    expect(screen).toBeTruthy();
  });
});
