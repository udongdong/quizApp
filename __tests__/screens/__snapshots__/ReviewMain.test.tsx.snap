// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewMain test render ReviewMain 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onGestureCancel={[Function]}
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          [
            {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            [
              {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <Text>
            리뷰 정보를 불러오고 있습니다.
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#5F37FF",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  review
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 60,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": [
          {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="review, tab, 1 of 1"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": true,
          }
        }
        accessibilityStates={
          [
            "selected",
          ]
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
            },
            {
              "backgroundColor": "transparent",
            },
            {
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "height": "100%",
                "marginTop": 3,
              },
              {
                "display": "none",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "color": "#5F37FF",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "color": "#8E8E8F",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            >
              ⏷
            </Text>
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              {
                "color": "#5F37FF",
              },
              {
                "fontSize": 13,
                "marginLeft": 20,
                "marginTop": 3,
              },
              {
                "fontSize": 15,
                "fontWeight": "700",
              },
            ]
          }
        >
          review
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
